etl/open311/open311_to_socrata.py

```diff
@@ -0,0 +1,169 @@
+"""
+Checks open311 API for new CSRs and updates a Socrata Dataset
+"""
+
+import argparse
+from datetime import datetime, timedelta, timezone
+from zoneinfo import ZoneInfo
+import os
+import logging
+import requests
+import time
+
+from sodapy import Socrata
+
+from etl import utils
+
+# Socrata Secrets
+SO_WEB = os.getenv("SO_WEB")
+SO_TOKEN = os.getenv("SO_TOKEN")
+SO_KEY = os.getenv("SO_KEY")
+SO_SECRET = os.getenv("SO_SECRET")
+REALTIME_DATASET = os.getenv("REALTIME_DATASET")
+
+# Open 311 API Key
+API_KEY = os.getenv("OPEN_311_API_KEY")
+API_BASE_URL = os.getenv("OPEN_311_API_BASE_URL")
+
+headers = {"Authorization": f"Bearer {API_KEY}"}
+
+# 10 requests per minute API limit
+MAX_REQUESTS_PER_MINUTE = 10
+SECONDS_PER_REQUEST = 60 / MAX_REQUESTS_PER_MINUTE
+
+# Timezone objects
+CENTRAL_TZ = ZoneInfo("America/Chicago")
+UTC_TZ = timezone.utc
+
+
+def determine_query_time(client, override_date=None):
+    """
+    Uses provided date arg first, and subtracts 10 minutes to reduce the risk we miss any updated records
+    If no arg is provided, the date used is the most recently updated socrata record.
+    If the dataset is empty, default to Oct 13, 2025.
+    """
+    if override_date:
+        try:
+            # Parse ISO 8601 datetime and subtract 10 minutes
+            dt = datetime.fromisoformat(override_date)
+            adjusted = dt - timedelta(minutes=10)
+            adjusted_str = adjusted.isoformat()
+            adjusted_str = adjusted_str.replace("+00:00", "Z")
+            logger.info(f"Using provided date (minus 10 minutes): {adjusted_str}")
+            return adjusted_str
+        except ValueError:
+            raise ValueError(
+                f"Invalid date format: {override_date}. Expected ISO 8601 format."
+            )
+
+    # get last updated record from socrata
+    date = client.get(
+        REALTIME_DATASET,
+        select="updated_datetime",
+        order="updated_datetime DESC",
+        limit=1,
+    )
+    if date:
+        # Interpret as Central Time and convert to UTC
+        ct_time = datetime.fromisoformat(date[0]["updated_datetime"]).replace(
+            tzinfo=CENTRAL_TZ
+        )
+        utc_time = ct_time.astimezone(UTC_TZ)
+        adjusted_str = utc_time.isoformat()
+        adjusted_str = adjusted_str.replace("+00:00", "Z")
+        return adjusted_str
+    return "2025-10-13T00:00:00Z"
```

Comment:
I understand what this date is -- a recent date that predates anything we might expect in the dataset, but can you please add a comment for future us who can't infer this? This is a magic number, and maybe unix timestamp 0 would be better, meaning Jan 1, 1970? as a "beginning of time" timestamp that most people will recognize and grok.